crates/amaru-ledger/src/rules.rs

```diff
@@ -22,6 +26,9 @@ pub enum BlockValidationError {
     RuleViolations(Vec<RuleViolation>),
     #[error("Cascading rule violations: root: {0:?}, resulting error(s): {1:?}")]
     Composite(RuleViolation, Box<BlockValidationError>),
+    // TODO: This error shouldn't exist, it's a placeholder for better error handling in less straight forward cases
+    #[error("Unnammed error: {0}")]
+    Unnamed(String),
```

Comment:
```suggestion
    #[error("Unnamed error: {0}")]
    Unnamed(String),
```
this is 100% nit, but I'd call it something like `UncategorizedError`, just so it was clear that it wasn't a validation error like "this block is missing a name"

Reply:
I like this a lot more